MdePkg/Include/Library/StandaloneMmDriverEntryPoint.h

```diff
@@ -45,6 +45,7 @@ extern CONST UINT8  _gDriverUnloadImageCount;
                                      ProcessModuleEntryPointList().
 
 **/
+NO_STACK_COOKIE

```

Comment:
How does a developer know when to apply 'NO_STACK_COOKIE' or not?

Reply:
There is a chicken and egg issue. Given the library updates the cookie when you return you will get a failure if the cookie has changed. We fixed this issue in our clang bespoke implementation by moving `_ModuleEntryPoint`() to `_ModuleCEntryPoint()` and making _ModuleEntryPoint an assembly gasket that writes __stack_chk_guard and calls  _ModuleCEntryPoint(). That way all the C code in the module has the same __stack_chk_guard/__security_cookie. We did not implement NO_STACK_COOKIE.

Reply:
I assume NO_STACK_COOKIE on the failure path is to prevent infinite loops of failure?

Reply:
I’ll add more to the documentation. This is only intended for scenarios where the architecture of the code prohibits stack cookies from being used, such as module entry points.

Reply:
Oh sorry, I was replying and it didn’t show Andrew’s responses until after I hit comment. I believe what he is saying is correct, I’ll follow up when I get back from vacation.

Reply:
Resolving as dropping the dynamic instance for now.